pkg/messaging/nats/pubsub.go

```diff
@@ -104,6 +114,10 @@ func (ps *pubsub) Subscribe(id, topic string, handler messaging.MessageHandler)
 		Subscription: sub,
 		cancel:       handler.Cancel,
 	}
+
+	if alreadySubscribed {
+		return ErrAlreadySubscribed
+	}
```

Comment:
You should not return an error. You only replace the old client with the existing one.